Storage/SQL/SQLitePasswords.swift

```diff
@@ -67,16 +67,18 @@ private class PasswordsTable<T>: GenericTable<Password> {
 
     override var factory: ((row: SDRow) -> Password)? {
         return { row -> Password in
-            let site = Site(url: row["hostname"] as String, title: "")
-            let pw = Password(site: site, username: row["username"] as String, password: row["password"] as String)
-            pw.httpRealm = row["httpRealm"] as String
-            pw.formSubmitUrl = row["formSubmitUrl"] as String
-            pw.usernameField = row["usernameField"] as String
-            pw.passwordField = row["passwordField"] as String
+            let site = Site(url: row["hostname"] as? String ?? "", title: "")
+            let pw = Password(site: site, username: row["username"] as? String ?? "", password: row["password"] as? String ?? "")
+
+            pw.httpRealm = row["httpRealm"] as? String ?? ""
+            pw.formSubmitUrl = row["formSubmitUrl"] as? String ?? ""
+            pw.usernameField = row["usernameField"] as? String ?? ""
+            pw.passwordField = row["passwordField"] as? String ?? ""
             pw.guid = row["guid"] as? String
-            pw.timeCreated = NSDate(timeIntervalSince1970: row["timeCreated"] as Double)
-            pw.timeLastUsed = NSDate(timeIntervalSince1970: row["timeLastUsed"] as Double)
-            pw.timePasswordChanged = NSDate(timeIntervalSince1970: row["timePasswordChanged"] as Double)
+            pw.timeCreated = NSDate(timeIntervalSince1970: row["timeCreated"] as? Double ?? 0)
+            pw.timeLastUsed = NSDate(timeIntervalSince1970: row["timeLastUsed"] as? Double ?? 0)
+            pw.timePasswordChanged = NSDate(timeIntervalSince1970: row["timePasswordChanged"] as? Double ?? 0)
```

Comment:
You used `as!` in most other places, but fall back to default values here. Why? Are you sure that this won't just paper over bugs and make them harder to find?